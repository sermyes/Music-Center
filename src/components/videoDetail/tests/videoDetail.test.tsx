import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import renderer from 'react-test-renderer';
import VideoDetail from './../videoDetail';
import { fakeVideo } from '../../tests/videos';

describe('Video Detail', () => {
  const video = fakeVideo;

  it('renders correctly', () => {
    const component = renderer.create(<VideoDetail video={video} />);

    expect(component.toJSON()).toMatchSnapshot();
  });

  it('renders video detail', () => {
    render(<VideoDetail video={video} />);

    const iframe = screen.getByTitle('youtube player')! as HTMLIFrameElement;
    const heading = screen.getByRole('heading');

    expect(iframe.src).toBe('https://www.youtube.com/embed/1');
    expect(heading.textContent).toBe(video.snippet.title);
  });
});
