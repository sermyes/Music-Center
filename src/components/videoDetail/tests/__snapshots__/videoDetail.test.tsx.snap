// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Video Detail renders correctly 1`] = `
<div
  className="container"
>
  <iframe
    allowFullScreen={true}
    className="video"
    src="https://www.youtube.com/embed/1"
    title="youtube player"
    width="100%"
  />
  <h2
    className="title"
  >
    title
  </h2>
</div>
`;
